File: java/src/main/java/vsdk/toolkit/material/RayHitFlag.java
package vsdk.toolkit.material;

public final class RayHitFlag {
    public static final int GEOMETRY = 0x01;
    public static final int PATCH = 0x02;
    public static final int POINT = 0x04;
    public static final int GEOMETRIC_NORMAL = 0x08;
    public static final int MATERIAL = 0x10;
    public static final int DISTANCE = 0x20;

    public static final int UV = 0x100;
    public static final int TEXTURE_COORDINATE = 0x200;
    public static final int SHADING_FRAME = 0x400;

    public static final int NORMAL = 0x800;

    public static final int FRONT = 0x10000;
    public static final int BACK = 0x20000;
    public static final int ANY = 0x40000;

    private RayHitFlag() {
    }
}

File: typescript/src/main/vsdk/toolkit/environment/geometry/elements/RayHitFlag.ts
export class RayHitFlag {
  public static readonly GEOMETRY = 0x01;
  public static readonly PATCH = 0x02;
  public static readonly POINT = 0x04;
  public static readonly GEOMETRIC_NORMAL = 0x08;
  public static readonly MATERIAL = 0x10;
  public static readonly DISTANCE = 0x20;

  public static readonly UV = 0x100;
  public static readonly TEXTURE_COORDINATE = 0x200;
  public static readonly SHADING_FRAME = 0x400;

  public static readonly NORMAL = 0x800;

  public static readonly FRONT = 0x10000;
  public static readonly BACK = 0x20000;
  public static readonly ANY = 0x40000;

  private constructor() {
  }
}
